presto-spi/src/main/java/com/facebook/presto/spi/InMemoryRecordSet.java

```diff
@@ -236,6 +237,10 @@ else if (VARCHAR.equals(type)) {
                     checkArgument(value instanceof String || value instanceof byte[],
                             "Expected value %d to be an instance of String or byte[], but is a %s", i, value.getClass().getSimpleName());
                 }
+                else if (VARBINARY.equals(type)) {
```

Comment:
If you need this, use `Slice` instead of `byte[]`. `byte[]` is here for `String` because it's old code.

But looks like you don't really need it.